Add solveOrderedRealCubic

EthanCurtiss opened the pull request.

Quenty reviewed (approved):
looks good to me, asking trey

Quenty commented on Modules/Shared/Math/PolynomialUtils.lua:
```diff
@@ -2,33 +2,81 @@
+		local z2 = coeff * math.cos(theta + math.pi / 6) - A/3
+		local solutions = table.create(3)
+		solutions[1], solutions[2], solutions[3] = z0, z1, z2
+		table.sort(solutions)
```
nit: maybe just use if-statements here.

Quenty reviewed (approved):
Looks good overall. I don't think you have merge permissions, so I'll just merge this, and you can put in another PR to fix the sorting if you want.

EthanCurtiss commented:
I'm pretty lost with this git stuff, but I think I the PR is correct now. Here are more details on what I changed

1. Switched cubeRoot to a much faster version by fractality
2. Quadratic calls Linear when a = 0
3. Quadratic returns 2 roots when there's only 1 solution with multiplicity 2
4. Cubic variables are named nicer
5. Swapped out table.sort in cubic for if statements
6. Assigned repeated calculations to variables in cubic
